usso/openid.go

```diff
@@ -78,15 +77,18 @@ func LoginHandler(w http.ResponseWriter, r *http.Request) {
 
 	resp, err := verify(url.String())
 	if err != nil {
+		// A mangled OpenID response is suspicious, so leave a nasty response
 		replyHTTPError(w, http.StatusBadRequest, err)
 		return
 	}
 
-	// get form username and get from datastore the User
+	// Check we have the mandatory fields in the OpenID response
 	username := r.Form.Get("openid.sreg.nickname")
-	if len(username) == 0 {
-		log.Println("Got no 'openid.sreg.nickname' in response params")
-		replyHTTPError(w, http.StatusBadRequest, errgo.New("OpenID response has not valid format"))
+	fullname := r.Form.Get("openid.sreg.fullname")
+	email := r.Form.Get("openid.sreg.email")
+	if len(username) == 0 || len(fullname) == 0 || len(email) == 0 {
```

Comment:
But the fullname in the frontend is used to identify if there is an authenticated user (evaluating if not null) and to show it in the navigation bar button, right?. In those both cases it doesn't matter if its value is the nickname or the full name; the important is having any value (is there any other use that needs the real full name and the nickname replacement is not valid?). The idea behind this is letting the user being authenticated even if he doesn't want to share more than the nickname